tools/crm_resource.c

```diff
@@ -532,9 +532,9 @@ agent_provider_cb(const gchar *option_name, const gchar *optarg, gpointer data,
 
 gboolean
 attr_set_type_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **error) {
-    if (crm_str_eq(option_name, "-m", TRUE) || crm_str_eq(option_name, "--meta", TRUE)) {
+    if (pcmk__str_eq(option_name, "-m", pcmk__str_none) || pcmk__str_eq(option_name, "--meta", pcmk__str_none)) {
         options.attr_set_type = XML_TAG_META_SETS;
-    } else if (crm_str_eq(option_name, "-z", TRUE) || crm_str_eq(option_name, "--utilization", TRUE)) {
+    } else if (pcmk__str_eq(option_name, "-z", pcmk__str_none) || pcmk__str_eq(option_name, "--utilization", pcmk__str_none)) {
```

Comment:
i guess it's another job for cocci :)

Reply:
Yeah, definitely.  One thing that's been bugging me is that pcmk__str_any_of and pcmk__str_none_of are case-insensitive and we do not have case-sensitive versions.  I'd like both so we can cover all these cases.  What do you think - add a flag, add another set of functions with different names, something else?

Reply:
The first idea that comes to my mind is rename pcmk__str_any_of to pcmk__str_any_of_i, do the same with pcmk__str_none_of, and write new functions with the old names that are case-insensitive.

Reply:
Logically, it makes sense for them to use pcmk__strcmp() and take the same flags, but that would make the calls long and a little weird-looking:

    pcmk__str_any_of(CRM_OP_REPROBE, operation, crm_op, NULL)
    ->
    pcmk__str_any_of(CRM_OP_REPROBE, pcmk__str_casei, operation, crm_op, NULL)

Another problem with that approach is that pcmk__str_null_matches wouldn't be usable since a NULL would be treated as the end of the list. pcmk__str_regex would work but there's probably no practical use case. So I think your suggestion is best. Maybe pcmk__strcase_any_of() to associate with strcasecmp()?

Reply:
Done.